api/v1beta2/addresspool_types.go

```diff
@@ -0,0 +1,74 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AddressPoolSpec defines the desired state of AddressPool.
+type AddressPoolSpec struct {
+	// A list of IP address ranges over which MetalLB has authority.
+	// You can list multiple ranges in a single pool, they will all share the
+	// same settings. Each range can be either a CIDR prefix, or an explicit
+	// start-end range of IPs.
+	Addresses []string `json:"addresses"`
+
+	// AutoAssign flag used to prevent MetallB from automatic allocation
+	// for a pool.
+	// +optional
+	// +kubebuilder:default:=true
+	AutoAssign *bool `json:"autoAssign,omitempty"`
+
+	// AvoidBuggyIPs prevents addresses ending with .0 and .255
+	// to be used by a pool.
+	// +optional
+	// +kubebuilder:default:=false
+	AvoidBuggyIPs bool `json:"avoidBuggyIPs,omitempty"`
+}
+
+// AddressPoolStatus defines the observed state of AddressPool.
+type AddressPoolStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
```

Comment:
is that a TODO?

Reply:
It's generated by controller-gen, and it was the same in the operator.